### ISSUE TITLE
Cannor run `simple build with a Docker container`: `No such service: builder`

### ISSUE THREAD

**Issue by nicokosi** [gratiartis/diagrams-as-code#4]
https://github.com/gratiartis/diagrams-as-code#simple-build-with-a-docker-container

```sh
$ uname -srv
Linux 5.8.0-59-generic #66~20.04.1-Ubuntu SMP Thu Jun 17 11:14:10 UTC 2021

$ git log -1
commit a93cbcf206a6bc4b4e96913527729a037b02a244 (HEAD -> main, origin/main, origin/HEAD)
Author: Stephen Masters <stephenm@digiterre.com>
Date:   Sat Jul 3 19:07:12 2021 +0100

    README improvements

$ docker-compose run --rm builder ./build-pdf.sh
ERROR: No such service: builder
```

I have also tried:
```sh
$ docker-compose run --rm docbuild

# :scissors: 
Step 9/9 : RUN npm install
 ---> Running in b1cbf4aca9a5
npm ERR! code EACCES
npm ERR! syscall open
npm ERR! path /documents/package-lock.json
npm ERR! errno -13
npm ERR! Error: EACCES: permission denied, open '/documents/package-lock.json'
npm ERR!  [Error: EACCES: permission denied, open '/documents/package-lock.json'] {
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   syscall: 'open',
npm ERR!   path: '/documents/package-lock.json'
npm ERR! }
npm ERR! 
npm ERR! The operation was rejected by your operating system.
npm ERR! It is likely you do not have the permissions to access this file as the current user
npm ERR! 
npm ERR! If you believe this might be a permissions issue, please double-check the
npm ERR! permissions of the file and its containing directories, or try running
npm ERR! the command again as root/Administrator.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/docuser/.npm/_logs/2021-07-05T06_26_16_633Z-debug.log
ERROR: Service 'docbuilder' failed to build : The command '/bin/sh -c npm install' returned a non-zero code: 243
```

**Comment by stephen-masters** [comment]
Ah! Sorry - that was me doing a stupid. I renamed the service in `docker-compose.yml` to be a bit more readable, and forgot to update the README which referred to it. The correct command is now:

```
docker-compose run --rm docbuilder ./scripts/build-pdf.sh
```

I have also modified the Dockerfile to use `npm ci` instead of `npm install`. This should prevent it from trying to modify `package-lock.json` during the build.

The Docker mechanism mounts the project directory into the container, and I think that is why the `package-lock.json` permissions are not being changed during the Docker image build. It's tempting to switch it to a multi-stage Dockerfile and copy the files in instead of mounting them. That might lead to a more reliable build process.
